src/server/req_rerun.c

```diff
@@ -358,16 +358,15 @@ timeout_rerun_request(struct work_task *pwt)
 	if ((pjob == NULL) || (pjob->ji_rerun_preq == NULL)) {
 		return;	/* nothing to timeout */
 	}
-
+	if (pjob->ji_rerun_preq->rq_conn != PBS_LOCAL_CONNECTION) {
+		conn_idx = connection_find_actual_index(pjob->ji_rerun_preq->rq_conn);
+	}
 	reply_text(pjob->ji_rerun_preq, PBSE_INTERNAL,
 		"rerunning job has timed out");
 
 	/* clear no-timeout flag on connection */
-	if (pjob->ji_rerun_preq->rq_conn != PBS_LOCAL_CONNECTION) {
-		conn_idx = connection_find_actual_index(pjob->ji_rerun_preq->rq_conn);
-		if (conn_idx != -1)
-			svr_conn[conn_idx].cn_authen &= ~PBS_NET_CONN_NOTIMEOUT;
-	}
+	if (conn_idx != -1)
```

Comment:
conn_idx is not initialized... So if it turns out to be a local connection then also you may end up getting -1 in conn_idx variable.
Can you please explain a little why this change is needed?